### PR TITLE
Add RSpec scenarios for Skype and Slack

### PR DESCRIPTION
**Pull request by bittner**
More scenarios for this fantastic Puppet module. Enjoy!

### CODE REVIEW

File: spec/classes/software_social_skype_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      end
+
+      context 'with defaults' do
+        if facts[:operatingsystem] == 'Darwin'
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_package(/Skype-.*+/).with_provider(appdmg) }
+        elsif facts[:operatingsystem] == 'Ubuntu'
+          it { is_expected.to compile.with_all_deps }
+          it {
+            is_expected.to contain_apt_source('skype-partner')
```
Comment (bittner) [inline spec/classes/software_social_skype_spec.rb]: There's clearly some syntax issue with `apt_source` here.

@edestecd Can you suggest how to reference `apt::source` with `contains_`?

File: spec/classes/software_social_skype_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts
+      end
+
+      context 'with defaults' do
+        if facts[:operatingsystem] == 'Darwin'
+          it { is_expected.to compile.with_all_deps }
+          it { is_expected.to contain_package(/Skype-.+/).with_provider('appdmg') }
```
Comment (bittner) [inline spec/classes/software_social_skype_spec.rb]: This step [is failing](https://travis-ci.org/edestecd/puppet-software/builds/290355530#L1112), too. I need to cover a [variable version string and the provider argument](https://github.com/edestecd/puppet-software/blob/master/manifests/social/skype.pp#L19-L23) (the regex and `with_provider` should do it).

@edestecd Can you suggest what could be the problem here? -- I've tried `with_provider(appdmg)` without quotes without success.

File: spec/classes/software_social_skype_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        elsif facts[:operatingsystem] == 'Ubuntu'
+          it { is_expected.to compile.with_all_deps }
+          it {
+            is_expected.to contain_apt_source('skype-partner')
```
Comment (bittner) [inline spec/classes/software_social_skype_spec.rb]: As before (now shown as "outdated" in the PR), there's clearly some syntax issue with `apt_source` here. Not sure though, why the [compiling into a catalogue](https://travis-ci.org/edestecd/puppet-software/builds/290355530#L1123-L1125) is failing.

@edestecd Can you suggest how to reference `apt::source` with `contains_`?

Comment (edestecd) [reply spec/classes/software_social_skype_spec.rb]: I think you do this: 
```ruby
# replace :: with __ in resource names
is_expected.to contain_apt__source('skype-partner')
```

File: spec/classes/software_social_skype_spec.rb (hunk 2 above)
Comment (edestecd) [reply spec/classes/software_social_skype_spec.rb]: I'm not sure. Have you tried it without the .with_provider ? If that still does not work, I might suspect the regex. I can't find any examples of using a regex in a resource name like that. You may have to hard code the version.

**Comment by bittner** [comment]
I've identified the culprit:

`software::params::skype_version` is only [defined for `Darwin` OSes](https://github.com/edestecd/puppet-software/blob/master/manifests/params.pp#L11).

I have to add this also the the `elsif 'Ubuntu'` clause.

File: spec/classes/software_social_skype_spec.rb (hunk 2 above)
Comment (bittner) [reply spec/classes/software_social_skype_spec.rb]: It's obviously the regex. So, I have to hardcode the version. :worried:

**Comment by bittner** [comment]
This PR is now read to merge. :sweat: :champagne: :beer:

File: .gitignore [hunk 4]
```diff
@@ -28,6 +28,7 @@ modules/
 spec/fixtures/
 .vagrant
 coverage/
+.rake_tasks~
```
Comment (edestecd) [inline .gitignore]: Should this have a ~?

**Comment by edestecd** [comment]
@bittner Looks excellent. I just had one comment about the gitignore

File: .gitignore (hunk 4 above)
Comment (edestecd) [reply .gitignore]: this is fine. Just making sure. Some editors like vi use ~ for line endings and it gets pasted by accident.

Comment (bittner) [reply .gitignore]: So, no changes needed? Ready for merging?  :smiley: :crossed_fingers:

**Comment by bittner** [comment]
Thanks, it was fun! :1st_place_medal: :champagne:

**Pull request merged**